File: abc127_c.py
# C - Prison
# https://atcoder.jp/contests/abc127/tasks/abc127_c

N, M = map(int, input().split())

lmax = 0
rmin = N
for i in range(M):
    L, R = map(int, input().split())
    lmax = max(L, lmax)
    rmin = min(R, rmin)

ans = max(rmin - lmax + 1, 0)
print(ans)

File: abc127_c.cpp
// https://atcoder.jp/contests/abc127/tasks/abc127_c
#include<iostream>
#include<algorithm>
//#include <numeric>
//#include<string>
//#include<vector>
//#include<map>
//#include<tuple>
//#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
using namespace std;

int main() {
	int N, M;
	cin >> N >> M;
	int L, R;
	int left = 1, right = N;
	for (int i = 0; i < M; i++)
	{

		cin >> L >> R;
		left = max(L, left);
		right = min(R, right);
	}

	int ans = max(right - left + 1, 0);

	cout << ans << endl;

	return 0;
}
